src/components/tokenassistant.js

```diff
@@ -0,0 +1,165 @@
+                this.tryLoadTokenAssistant();
+            }, 20);
+        }
+    }
+
+    addScriptToIndexFile = () => {
+        var head = window.document.head;
+        var script = window.document.createElement('script');
+        script.type = 'text/javascript';
+        script.src = this.config.assisted_token_endpoint +
+            '/resources/js/assisted-token.js';
+        script.id = 'assisted-token-js-script';
+        head.appendChild(script);
+    }
+
+    loadTokenAssistant = () => {
+        if (!window.curity) {
+            throw new Error('Assisted token javascript was not found.' +
+                ' Make sure the server is running and/or update URL ' +
+                'of #assisted-token-js-script script');
+        }
+        window.curity.debug = this.debug;
+        this.tokenAssistant = window.curity.token.assistant({
+            clientId: this.clientId,
+        });
+    }
+
+    getToken = () => {
+        if (!this.tokenAssistant) {
+            console.error('Token Assistant is undefined.');
+            return false;
+        }
+        this.tokenAssistant.loginIfRequired({ "scope": this.scope }).then((msg) => {
+            console.log('Retrieved tokens', msg);
+            let additionalData = this.tokenAssistant.getAdditionalData();
+            console.log('additional data', additionalData)
+
+            this.setState({ isLoggedIn: true, accessToken: msg.token, userName: additionalData.subject });
+        }).fail((err) => {
+            console.log('Failed to retrieve tokens', err);
+        });
+    }
+
+    logout = () => {
+        if (!this.tokenAssistant) {
+            console.error('Token Assistant is undefined.');
+            return false;
+        }
+        this.tokenAssistant.logout().then(() => {
+            this.setState({ accessToken: undefined, userName: undefined, isLoggedIn: false });
+        }).fail((err) => {
+            console.err("Failed to logout.", err);
+        });
+    }
+}
+
+const StateBox = (props) => {
+    return (
+        <Consumer>
+            {(context) => context.isLoggedIn ? <pre>{context.accessToken}</pre> : ''}
+        </Consumer>
+    )
+}
+
+const Greeting = (props) => {
+    return (
+        <Consumer>
+            {context => context.userName !== undefined ? <h1>Hi {context.userName}</h1> : <h1>Welcome!</h1>}
+        </Consumer>
+    )
+}
+
+const ControlPanel = () => {
+    return (
+        <Consumer>
+            {(context) => {
+                const loginButton = !context.isLoggedIn ? <button onClick={context.login}>Login</button> : '';
+                const logoutButton = context.isLoggedIn ? <button onClick={context.logout}>Logout</button> : ''
+                const getTokenButton = context.isLoggedIn ? <button onClick={context.login}>Get Token</button> : '';
+                return (<p>{loginButton} {logoutButton} {getTokenButton}</p>);
+            }
+            }
+        </Consumer>
+    )
+}
+
+function TokenAssistant(props) {
```

Comment:
any reason to use function here and not `const TokenAssistant = (props) =>` as above for consistency?

Reply:
No, I'll update to make it consistent.